cpp/src/parquet/column_reader.cc

```diff
@@ -1478,13 +1478,14 @@ class TypedRecordReader : public TypedColumnReaderImpl<DType>,
     // We skipped the levels by incrementing 'levels_position_'. For values
     // we do not have a buffer, so we need to read them and throw them away.
     // First we need to figure out how many present/not-null values there are.
-    std::shared_ptr<::arrow::ResizableBuffer> valid_bits;
-    valid_bits = AllocateBuffer(this->pool_);
-    PARQUET_THROW_NOT_OK(valid_bits->Resize(bit_util::BytesForBits(skipped_records),
-                                            /*shrink_to_fit=*/true));
+    if (valid_bits_ == nullptr) {
+      valid_bits_ = AllocateBuffer(this->pool_);
+    }
+    PARQUET_THROW_NOT_OK(valid_bits_->Resize(bit_util::BytesForBits(skipped_records),
+                                             /*shrink_to_fit=*/true));
```

Comment:
Why it's set to true? 🤔

Reply:
got the same question too. i think the original motivation is just to be conservative on memory usage. We could revisit if setting to false is better. Rightnow i just keep it as true.